### PR TITLE
CATROID-1192 Fix of LookFragmentActivityRecreateRegressionTest

### PR DESCRIPTION
**Pull request by csum1**
simple fix of testActivityRecreateRenameLookDialog

Ticket: [CATROID-1192](https://jira.catrob.at/browse/CATROID-1192)

- Conversion of LookFragmentActivityRecreateRegressionTest to Kotlin
- The failed testcase used older mechanism => when renaming/copy/etc. and there is only one item (in this case a Look) then this one is chosen automatically. The user does not have to do this step in between to chose the item. This bug was introduced with [CATROID-885](https://jira.catrob.at/browse/CATROID-885)
- There are some other tests that rely on the old mechanism and therefore fail

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/regression/activitydestroy/LookFragmentActivityRecreateRegressionTest.kt [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        createProject()
+        baseActivityTestRule.launchActivity()
+    }
+
+    @Category(AppUi::class, Smoke::class, Quarantine::class)
+    @Flaky
```
Comment (bengaal) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/regression/activitydestroy/LookFragmentActivityRecreateRegressionTest.kt:84]: I think you can remove the `@Flaky` tags for the tests. I run the tests locally multiple times with: 
```
class RepeatRule : TestRule {

    private class RepeatStatement(private val statement: Statement, private val repeat: Int) : Statement() {
        @Throws(Throwable::class)
        override fun evaluate() {
            for (i in 0..repeat - 1) {
                statement.evaluate()
            }
        }
    }

    override fun apply(statement: Statement, description: Description): Statement {
        var result = statement
        val repeat = description.getAnnotation(RepeatTest::class.java)
        if (repeat != null) {
            val times = repeat.value
            result = RepeatStatement(statement, times)
        }
        return result
    }
}
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.ANNOTATION_CLASS)
annotation class RepeatTest(val value: Int = 1)
```
and they don't seem to produce inconsistent runs.

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/regression/activitydestroy/LookFragmentActivityRecreateRegressionTest.kt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            .inRoot(RootMatchers.isDialog())
+            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+
+        InstrumentationRegistry.getInstrumentation()
+            .runOnMainSync { baseActivityTestRule.activity.recreate() }
+        InstrumentationRegistry.getInstrumentation().waitForIdleSync()
+    }
+
+    @Category(AppUi::class, Smoke::class, Quarantine::class)
+    @Flaky
```
Comment (bengaal) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/regression/activitydestroy/LookFragmentActivityRecreateRegressionTest.kt:100]: See above

**Review by bengaal — APPROVED** [review]
Please consider implementing my feedback.
In general it looks good to me.

**Comment by wslany** [comment]
Please can someone confirm that the test passes? Unfortunately it is never executed on Jenkins AFAIK.

**Comment by wslany** [comment]
Thanks!!

**Pull request merged**